ClientLibrary/PsiphonTunnel.go

```diff
@@ -123,27 +140,30 @@ func psiphon_tunnel_start(cConfigJSON, cEmbeddedServerEntryList, cClientPlatform
 
 	config, err := psiphon.LoadConfig([]byte(configJSON))
 	if err != nil {
-		return startErrorJson(err)
+		managedStartResult = startErrorJson(err)
+		return managedStartResult
 	}
 
 	// Set network ID
 
-	if networkID != "" {
-		// Ensure config.NetworkID is on the Go heap
-		config.NetworkID = deepCopy(networkID)
-	}
+	config.NetworkID = networkID
 
 	// Set client platform
 
-	if clientPlatform != "" {
-		// Ensure config.ClientPlatform is on the Go heap
-		config.ClientPlatform = deepCopy(clientPlatform)
+	config.ClientPlatform = clientPlatform
+
+	// Set timeout
+
+	if timeout != nil {
+		// timeout overrides optional timeout field in config
+		config.EstablishTunnelTimeoutSeconds = nil
```

Comment:
I think this needs to be:
```
config.EstablishTunnelTimeoutSeconds = 0
```

If you set it to `nil`, the _default_ value will be used, which might be shorter than `timeout`. Setting it to `0` will disable the timeout within tunnel-core.